packages/packages/commonwell-sdk/src/client/commonwell.ts

```diff
@@ -768,6 +768,23 @@ export class CommonWell implements CommonWellAPI {
     return document.query(this.api, headers, patientId);
   }
 
+  /**
+   * Queries a patient's Documents - including other possible results.
+   *
+   * @param meta       Metadata about the request.
+   * @param patientId  The patient's ID.
+   * @returns {Promise<DocumentQueryFullResponse>}
+   * @see {@link https://specification.commonwellalliance.org/services/data-broker/cha-broker-api-reference#104-document-query|Use case}
+   * @see {@link https://specification.commonwellalliance.org/services/data-broker/protocol-operations-data-broker#8781-find-documents|API spec}
+   */
+  async queryDocumentsFull(
+    meta: RequestMetadata,
+    patientId: string
+  ): Promise<DocumentQueryFullResponse> {
+    const headers = await this.buildQueryHeaders(meta);
+    return document.queryFull(this.api, headers, patientId);
+  }
```

Comment:
What do you think we add an optional `options` param to `queryDocuments` to indicate whether it should return only documents (filtering out invalid ones) or include errors/operation outcomes as well, something like:

```
queryDocuments(meta: RequestMetadata, patientId: string, options?: { includeFailures: boolean })
```

It's easier from the SDK user's perspective to have one function to return docs and just parameterize it accordingly. I understand we'd have a diff return type on `queryDocuments`, but I think its worth it.

Reply:
To clarify the intention here - in a subsequent PR I'll remove the one that only gets doc refs, as I don't see a good use case for it. Just wanted to get this out the door quicker before also doing a bit of a refactoring on the API.

Reply:
Sounds good.